src/com/xilinx/rapidwright/MainEntrypoint.java

```diff
@@ -0,0 +1,182 @@
+import com.xilinx.rapidwright.router.Router;
+import com.xilinx.rapidwright.tests.CheckAccuracyUsingGnlDesigns;
+import com.xilinx.rapidwright.tests.DeviceLoader;
+import com.xilinx.rapidwright.tests.PinMapTester;
+import com.xilinx.rapidwright.tests.ReportDevicePerformance;
+import com.xilinx.rapidwright.util.BrowseDevice;
+import com.xilinx.rapidwright.util.CompareRouteStatusReports;
+import com.xilinx.rapidwright.util.DesignImplementationDiff;
+import com.xilinx.rapidwright.util.FileTools;
+import com.xilinx.rapidwright.util.Installer;
+import com.xilinx.rapidwright.util.JobQueue;
+import com.xilinx.rapidwright.util.PartPrinter;
+import com.xilinx.rapidwright.util.PerformanceExplorer;
+import com.xilinx.rapidwright.util.RapidWright;
+import com.xilinx.rapidwright.util.StringTools;
+import com.xilinx.rapidwright.util.Unzip;
+
+public class MainEntrypoint {
+    interface MainStyleFunction<E extends Throwable> {
+        void main(String[] args) throws E;
+    }
+
+    private static final Map<String, MainStyleFunction<?>> functions = new HashMap<>();
+    private static final List<String> functionNames = new ArrayList<>();
+
+    private static void addFunction(String name, MainStyleFunction<?> func) {
+        functions.put(name.toLowerCase(), func);
+        functionNames.add(name);
+    }
+
+    static {
+        addFunction("AddSubGenerator", AddSubGenerator::main);
+        addFunction("BlockCreator", BlockCreator::main);
+        addFunction("BlockStitcher", BlockStitcher::main);
+        addFunction("BlockUpdater", BlockUpdater::main);
+        addFunction("BrowseDevice", BrowseDevice::main);
+        addFunction("CheckAccuracyUsingGnlDesigns", CheckAccuracyUsingGnlDesigns::main);
+        addFunction("CompareRouteStatusReports", CompareRouteStatusReports::main);
+        addFunction("CopyMMCMCell", CopyMMCMCell::main);
+        addFunction("CustomRouting", CustomRouting::main);
+        addFunction("DecomposeLUT", DecomposeLUT::main);
+        addFunction("DesignImplementationDiff", DesignImplementationDiff::main);
+        addFunction("DesignInstrumentor", DesignInstrumentor::main);
+        addFunction("DeviceBrowser", DeviceBrowser::main);
+        addFunction("DeviceLoader", DeviceLoader::main);
+        addFunction("DeviceResourcesExample", DeviceResourcesExample::main);
+        addFunction("EDIFNetlist", EDIFNetlist::main);
+        addFunction("EDIFParser", EDIFParser::main);
+        addFunction("EDIFPropertyValue", EDIFPropertyValue::main);
+        addFunction("EDIFTools", EDIFTools::main);
+        addFunction("EnumerateCellBelMapping", EnumerateCellBelMapping::main);
+        addFunction("ExampleNetlistCreation", ExampleNetlistCreation::main);
+        addFunction("FileTools", FileTools::main);
+        addFunction("HandPlacer", HandPlacer::main);
+        addFunction("ILAInserter", ILAInserter::main);
+        addFunction("ImplGuide", ImplGuide::main);
+        addFunction("Installer", Installer::main);
```

Comment:
@jakobwenzel Let's remove the Installer, I can't think of any useful cases where we would want to run this from an existing install.

Reply:
Done